### DISCUSSION TITLE
@mantine/tag-picker -- a few ideas/suggestions

### DISCUSSION THREAD

**Discussion by cstrat** [mantinedev/mantine#85] [Feedback]
1. **Let us use a React element instead of the `<Badge>`**
It would be nice if we could pass a react element in to be used as the trigger and display for the tag-picker.
For example, to use a `<Button>` instead of the badge.
The props should still pass through so the button text can read the `value` etc...

2. **Give us more control of the dropdown/popup**
Right now in one of my apps I have 24 fixed options. I want to use the tag picker as a glorified select box because it looks nicer and has filtering which can help if that list of 24 grows.
However - unless I missed it, I have no way to limit the popup size, ideally I only want 5 or 6 options displayed. The user should be able to either scroll or just filter via the text entry.

**Comment by cstrat** [comment]
I forgot to add on point 2, **the position of the popup!**
Right now it pops up to the right. In my application this causes a horizontal scrollbar to appear because the tag-picker is to the right of the screen.
This jerks the UI around a bit and when it closes it happens again.

**Comment by cstrat** [comment]
It also appears to be missing a `disabled` prop.

**Comment by cstrat** [comment]
I also find it strange that the `value` needs to be the entire object:

```
{
  id: 'xxx',
  name: 'xxx',
  color: 'xxx',
}
```

And not simply the value of the `id`...

**Comment by rtivital** [comment]
That's all true and nice to have. I've built tag picker initially for my budget tracking application and it has very limited usage. I'll consider to add more features to it in future releases but right now I want to focus on more essential components in @mantine/core and upcoming @mantine/dates packages.

**Reply by cstrat** [reply]
Yeah of course!

Thought I would just put the suggestions out here.
Certainly not a first priority.

**Reply by rtivital** [reply]
Sure, thanks for suggestions, will do something about this package later
